pkgs/applications/video/obs-studio/v4l2sink.nix

```diff
@@ -1,42 +1,49 @@
-{ stdenv, fetchFromGitHub
-, cmake, pkgconfig, wrapQtAppsHook
-, obs-studio }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, qt5
+, obs-studio
+}:
 
-stdenv.mkDerivation {
-  pname = "obs-v4l2sink-unstable";
-  version = "20181012";
+stdenv.mkDerivation rec {
+  pname = "obs-v4l2sink";
+  version = "0.1.0";
 
   src = fetchFromGitHub {
     owner = "CatxFish";
     repo = "obs-v4l2sink";
-    rev = "1ec3c8ada0e1040d867ce567f177be55cd278378";
-    sha256 = "03ah91cm1qz26k90mfx51l0d598i9bcmw39lkikjs1msm4c9dfxx";
+    rev = version;
+    sha256 = "0l4lavaywih5lzwgxcbnvdrxhpvkrmh56li06s3aryikngxwsk3z";
   };
 
-  nativeBuildInputs = [ cmake pkgconfig wrapQtAppsHook ];
-  buildInputs = [ obs-studio ];
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ qt5.qtbase.dev obs-studio ];
 
   patches = [
-    ./0001-find-ObsPluginHelpers.cmake-in-the-obs-src.patch
+    ./fix-v4l2device_close-segfault.patch
```

Comment:
This should be a compile error, not segfault, if I am reading the patch right.

Moreover could you use `fetchpatch` to obtain `https://github.com/CatxFish/obs-v4l2sink/commit/6604f01796d1b84a95714730ea51a6b8ac0e450b.patch` instead of adding it in-tree? And also add a link to the PR in a comment.

Reply:
The compile error results in a segfault (if you look at the related issues on the issue tracker).
I've personally had issues with non-deterministic patch/diff output from GitHub (changing hashes). Is it OK if we keep it as a vendored patch since the next version is probably going to include it anyway?

Reply:
Oh, it leads to UB. :woman_shrugging:

Reply:
`fetchpatch` is specifically designed to make the patches more deterministic by cleaning the git fluff that changes. Looking at the commit history, I am not sure there will ever be a next release.